### PR TITLE
fix: annoying mypy error is annoying

### PR DESCRIPTION
**Pull request by pauldambra**
lately I get an error when running mypy

i'm not the only one: https://posthog.slack.com/archives/C0113360FFV/p1733155706603479

`AssertionError: Cannot find component 'trace' for 'opentelemetry.trace'`

on updating to latest mypy I then get

`OverflowError: int too big to convert`

We can fix this by not writing to a mypy cache. Since that's what's failing

This is a workaround not a fix 🙈 

----

Regardless of how I try to ignore it the last output we see in verbose mode is something like

```
LOG:  Cached module mimesis.providers.payment has same interface
LOG:  Writing mimesis.providers.internet /Users/paul/github/posthog/env/lib/python3.11/site-packages/mimesis/providers/internet.py mimesis/providers/internet.meta.json mimesis/providers/internet.data.json
LOG:  Build finished in 8.507 seconds with 7108 modules, and 0 errors
OverflowError: int too big to convert
```

if I delete the `py.typed` file in the mimesis package then we no longer get the error but things like

```
[mypy-mimesis.providers.*]
ignore_errors = True
```

in the mypy.ini didn't change anything

presumably mimesis has a very large number in it (or something that resolve to one) and when mypy tries to cache it the json serializer can't cope 🤷 

i tried updating to mimesis v18 which didn't affect anything - and is a 13 major version jump so I didn't include it here

it isn't clear to me if this is a bug in mypy or mimesis or the overlap between them (or indeed something else 🙈)

### CODE REVIEW

**Comment by pauldambra** [comment]
downgrading to 

```
mypy~=1.7.1
django-stubs==4.2.7
```

means no more errors _running_ mypy
but hundreds more violations reported

**Review by ioannisj — APPROVED** [review]
Low context but makes sense

**Comment by pauldambra** [comment]
ok, this is silly. mypy results are apparently bouncing but 🤷 

going to binary search in commits using `uv pip install -r requirements.txt -r requirements-dev.txt && rm -rf .mypy_cache && mypy -p posthog | mypy-baseline filter && mypy -p posthog | mypy-baseline filter` see if i can narrow down when things went wrong

File: mypy-baseline.txt
Comment (Twixes) [inline mypy-baseline.txt:1]: Where did this go? 🤯

Comment (pauldambra) [reply mypy-baseline.txt:1]: completely tore down my environment, and still get this result 🤷

Comment (pauldambra) [reply mypy-baseline.txt:1]: i don't get the error today, but i've installed so many different versions of mypy and associated deps that i don't trust my machine

**Pull request closed without merging**